File: Helpers/src/main/java/com/dg/helpers/IntHelper.java
package com.dg.helpers;

/**
 * Created by Daniel Cohen Gindi (danielgindi@gmail.com)
 */
public class IntHelper
{
    public static Integer withObject(Object value)
    {
        if (value == null) return null;
        else if (value instanceof Integer) return (Integer)value;
        else if (value instanceof Short) return (int) (Short) value;
        else if (value instanceof String)
        {
            try { return Integer.parseInt((String) value); }
            catch (Exception ignored) { }
        }
        return null;
    }

    public static Integer withObject(Object value, int defaultValue)
    {
        if (value == null) return defaultValue;
        else if (value instanceof Integer) return (Integer)value;
        else if (value instanceof Short) return (int) (Short) value;
        else if (value instanceof String)
        {
            try { return Integer.parseInt((String) value); }
            catch (Exception ignored) { }
        }
        return defaultValue;
    }

    public static int[] toArray(Integer[] inArray)
    {
        int[] array = new int[inArray.length];
        for (int i = 0, len = inArray.length; i < len; i++)
        {
            Integer value = inArray[i];

            if (value != null)
            {
                array[i] = value;
            }
        }
        return array;
    }
}

File: Helpers/src/main/java/com/dg/helpers/IntHelper.kt
package com.dg.helpers

@Suppress("unused")
object IntHelper
{
    @Suppress("MemberVisibilityCanBePrivate")
    fun withObject(value: Any?): Int?
    {
        when (value)
        {
            null -> return null
            is Int -> return value
            is Short -> return value.toInt()
            is String -> try
            {
                return Integer.parseInt(value)
            }
            catch (ignored: Exception)
            {
            }
        }
        return null
    }

    fun withObject(value: Any?, defaultValue: Int): Int
    {
        return withObject(value) ?: defaultValue
    }

    fun toArray(inArray: Array<Int?>): IntArray
    {
        val array = IntArray(inArray.size)
        var i = 0
        val len = inArray.size
        while (i < len)
        {
            val value = inArray[i]

            if (value != null)
            {
                array[i] = value
            }
            i++
        }
        return array
    }
}
